### PR TITLE
Issue #17: Sphinx documentation

### PR DESCRIPTION
**Pull request by johnowagon**
## Description
Part of #17 

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

How should this pull request be reviewed?
- [X] By commit
- [ ] All changes at once

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [ ] __Unit tests__ (General Environment only) `pytest --cov bsk_rl/envs/general_satellite_tasking --cov-report term-missing tests/unittest`
- [ ] __Integrated tests__ (General Environment only) `pytest --cov bsk_rl/envs/general_satellite_tasking --cov-report term-missing tests/integration`

Does not change code functionality

### Test Configuration
 - Python: [3.10.1]
-  Basilisk: [2.2.0]
 - Platform: [OSX 12.6.6]

# Checklist:

- [X] My code follows the style guidelines of this project (passes Black, ruff, and isort)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Commit messages are atomic, are in the form `Issue #XXX: Message` and have a useful message
- [x] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

To test locally since auto-generated documentation files are not being tracked by git:
From project root:

```
sphinx-apidoc -e -f -o docs/source/documentation src/bsk_rl/
cd docs && make html
```

The output pages can be viewed from `docs/build/html/index.html`. Note that some pieces of documentation may look wrong, this is due to some syntax errors in docstrings, #94 addresses and fixes these issues.

### CODE REVIEW

**Comment by Mark2000** [comment]
Updated to not use sphinx-apidoc for more flexibility. Currently running `make html` in the docs folder should build everything.

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,19 @@
+cff-version: 1.2.0
+message: "If you use this software, please cite it as below."
+authors:
+- family-names: "Stephenson"
+  given-names: "Mark"
+  # orcid: "https://orcid.org/0000-0000-0000-0000"
```
Comment (LorenzzoQM) [inline CITATION.cff]: Should we add the ORCIDs?

Comment (Mark2000) [reply CITATION.cff]: Sure, I should probably get around to making one. Feel free to send me yours

File: docs/Makefile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+clean:
+	rm -rf "source/Examples" "source/API Reference"
+
+
+view:
+	# works on macOS
+	open build/html/index.html
```
Comment (LorenzzoQM) [inline docs/Makefile]: Add a last line so GitHub does get mad at us

File: docs/source/publications.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+#. \M. Stephenson and H. Schaub, `“Optimal Target Sequencing In The Agile Earth-Observing Satellite Scheduling Problem Using Learned Dynamics,” <https://hanspeterschaub.info/Papers/Stephenson2023.pdf>`_ AAS/AIAA Astrodynamics Specialist Conference, Big Sky, Montana, August 13–17 2023. 
+#. \M. Stephenson, L. Quevedo Manotovani, S. Phillips and H. Schaub, `“Using Enhanced Simulation Environments to Accelerate Reinforcement Learning for Long-Duration Satellite Autonomy,” <https://hanspeterschaub.info/Papers/Stephenson2024.pdf>`_ AIAA Science and Technology Forum and Exposition (SciTech), Orlando, Florida, Jan. 8–12, 2024. doi:10.2514/6.2024-0990.
+#. \M. Stephenson and H. Schaub, “Autonomous Earth-Observing Satellite Scheduling Using Reinforcement Learning With Event-Based Decision Intervals,” AAS Rocky Mountain GN&C Conference, Breckenridge, CO, Feb. 2–7, 2024.
+#. \M. Stephenson, L. Quevedo Mantovani and H. Schaub, “Intent Sharing and Auctions for Emergent Collaboration in Autonomous Earth Observing Constellations,” AAS Rocky Mountain GN&C Conference, Breckenridge, CO, Feb. 2–7, 2024.
```
Comment (LorenzzoQM) [inline docs/source/publications.rst]: We need to update the title of the last conference paper

File: examples/plotting_tools/plot_a2c_hyperparams.py [hunk 4]
```diff
@@ -4,7 +4,8 @@
 import matplotlib as mpl
 import numpy as np
 from matplotlib import pyplot as plt
-from plot_dqn_hyperparams import concatenate_results
+
```
Comment (LorenzzoQM) [inline examples/plotting_tools/plot_a2c_hyperparams.py:7]: Just curious. Why the extra space?

Comment (Mark2000) [reply examples/plotting_tools/plot_a2c_hyperparams.py:7]: Those imports were broken, fixing the import the causes ruff/isort to put it seepage cause its a local import, which those tools group separately

Comment (LorenzzoQM) [reply examples/plotting_tools/plot_a2c_hyperparams.py:7]: Interesting. Thanks for the explanation

**Comment by Mark2000** [comment]
@LorenzzoQM addressed your comments and ready for approval

**Review by LorenzzoQM — APPROVED** [review]

**Comment by LorenzzoQM** [comment]
Thanks for making those changes. I just approved the pull request. This documentation is an excellent addition to the repo!

**Pull request merged**